app/models/taxon.rb

```diff
@@ -57,6 +57,10 @@ def base_path=(base_path)
     @path_slug = path_components['slug']
   end
 
+  def legacy_taxons=(legacy_taxons)
+    @legacy_taxons = legacy_taxons.select(&:present?)
```

Comment:
is this necessary? not-present legacy taxons seem to be removed before calling this?, i.e. links_update:7